http.go

```diff
@@ -1,17 +1,142 @@
 package veneur
 
 import (
+	"compress/zlib"
+	"encoding/json"
+	"hash/fnv"
+	"io"
 	"net/http"
+	"sort"
+	"time"
 
+	"github.com/Sirupsen/logrus"
 	"goji.io"
 	"goji.io/pat"
 	"golang.org/x/net/context"
 )
 
 func (s *Server) Handler() http.Handler {
 	mux := goji.NewMux()
+
 	mux.HandleFuncC(pat.Get("/healthcheck"), func(c context.Context, w http.ResponseWriter, r *http.Request) {
 		w.Write([]byte("ok\n"))
 	})
+
+	mux.HandleFuncC(pat.Post("/import"), func(c context.Context, w http.ResponseWriter, r *http.Request) {
+		start := time.Now()
+
+		var (
+			jsonMetrics []JSONMetric
+			body        io.ReadCloser
+			err         error
+		)
+		switch enc := r.Header.Get("Content-Encoding"); enc {
+		case "":
+			body = r.Body
+		case "deflate":
+			body, err = zlib.NewReader(r.Body)
+			if err != nil {
+				http.Error(w, err.Error(), http.StatusBadRequest)
+				s.logger.WithFields(logrus.Fields{
+					"encoding": enc,
+					"client":   r.RemoteAddr,
+				}).Error("Could not read compressed request body")
+				s.statsd.Count("import.request_error_total", 1, nil, 1.0)
+				return
+			}
+			defer body.Close()
```

Comment:
overall I would say it's a matter of preference. I usually assume that if I couldn't open the reader, then I won't be able to close it either, so I won't bother actually attempting to close unless I know the creation of the reader was a success. (most closers actually return an error, but that error is usually unactionable so it's not worth the extra trouble to capture it in the deferred function)